File: frontend/playwright/runner.py
import pytest
import os
import sys

tests_dir = os.path.join(os.path.dirname(__file__), "tests")

tests = [
    os.path.join(tests_dir, "test_queue.py")
]

code = 0

for test in tests:
    pytest_cmd = ["-v", test]
    exit_code = pytest.main(pytest_cmd)
    if exit_code != 0:
        code = 1

sys.exit(code)

File: frontend/playwright/runner.js
import { spawn } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function runPython(script, args = []) {
    const scriptPath = path.resolve(__dirname, script);

    const p = spawn('pytest', [scriptPath, ...args], { stdio: 'inherit' });

    p.on('error', (err) => {
        console.error(`Error: ${err}`);
    });
}

function run() {
    runPython('tests/test_queue.py');
}

run()


